packages/cli/src/util/login/saml.ts

```diff
@@ -1,14 +1,55 @@
 import { URL } from 'url';
 import type Client from '../client';
+import { inspectTokenRequest, processInspectTokenResponse } from '../oauth';
 import doOauthLogin from './oauth';
 
-export default function doSamlLogin(
+export default async function doSamlLogin(
   client: Client,
-  teamIdOrSlug: string,
+  team: { teamIdOrSlug: string; slug: string },
   outOfBand?: boolean,
   ssoUserId?: string
 ) {
+  if (client.authConfig.type === 'oauth') {
+    const sessionId = await getSessionId(client);
+    const url = new URL(
+      `https://vercel.com/sso/${team.slug}?${new URLSearchParams({
+        session_id: sessionId,
+      }).toString()}`
+    );
+    return doOauthLogin(
+      client,
+      url,
+      'SAML Single Sign-On',
+      outOfBand,
+      ssoUserId
+    );
+  }
   const url = new URL('/auth/sso', client.apiUrl);
-  url.searchParams.set('teamId', teamIdOrSlug);
+  url.searchParams.set('teamId', team.teamIdOrSlug);
   return doOauthLogin(client, url, 'SAML Single Sign-On', outOfBand, ssoUserId);
 }
+
+export async function getSessionId(client: Client): Promise<string> {
```

Comment:
Nit: return both session ID and client ID here?
```suggestion
export async function getSessionAndClientId(client: Client): Promise<{ session_id: string, client_id: string }> {
```